### PR TITLE
Use random namespaces

### PR DESCRIPTION
**Pull request by ironcladlou**
Use a random namespace for each test. Simplify the config.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,17 +17,12 @@ Here's an example config file:
...
 listenAddress: "127.0.0.1:8080"
-check:
-  namespace: example
-  displayName: Workspace for monitor-project-lifecycle Test
-  route: django-psql-persistent
-runInterval: "1m"
-timeout:
-  templateCreation: "10m"
-  templateDeletion: "5m"
+runInterval: 1m
```
Comment (derekwaynecarr) [inline README.md:20]: config should be versioned.

File: cmd/monitor/monitor.go [hunk 2]
```diff
@@ -210,38 +212,40 @@ func (m *AppCreateAvailabilityMonitor) runTest() error {
...
 	return nil
 }
 
-// setupWorkspace creates a project to hold an individual test's resources.
+// createProject creates a project to hold an individual test's resources.
+// Returns the generated name of the project.
 // TODO: Add prometheus metrics for timing and errors.
-func (m *AppCreateAvailabilityMonitor) setupWorkspace() error {
+func (m *AppCreateAvailabilityMonitor) createProject() (string, error) {
+	name := "monitor-app-create-" + utilrand.String(4)
```
Comment (derekwaynecarr) [inline cmd/monitor/monitor.go:225]: per discussion, we should label to know if a project from a previous run is still hanging around, and fail a next test run if the namespace was not properly deleted last run.  i do not want a random bug to drive 10k namespaces appearing on the cluster accidentally.

File: cmd/monitor/monitor.go [hunk 3]
```diff
@@ -210,38 +212,40 @@ func (m *AppCreateAvailabilityMonitor) runTest() error {
...
 }
 
-// cleanupWorkspace deletes the project for a test.
+// deleteProject deletes the project for a test.
 // TODO: Add prometheus metrics for timing and errors.
-func (m *AppCreateAvailabilityMonitor) cleanupWorkspace() error {
+func (m *AppCreateAvailabilityMonitor) deleteProject(namespace string) error {
 	// Delete the project that contains the kube resources we've created
-	err := m.Client.ProjectClient.Projects().Delete(m.Config.Check.Namespace, &metav1.DeleteOptions{})
+	err := m.Client.ProjectClient.Projects().Delete(namespace, &metav1.DeleteOptions{})
```
Comment (derekwaynecarr) [inline cmd/monitor/monitor.go:246]: wait and verify the namespace is actually removed.

**Review by derekwaynecarr — APPROVED** [review]
please address comments in follow-up

File: README.md (hunk 1 above)
Comment (ironcladlou) [reply README.md:20]: https://github.com/openshift/monitor-project-lifecycle/issues/23

File: cmd/monitor/monitor.go (hunk 2 above)
Comment (ironcladlou) [reply cmd/monitor/monitor.go:225]: https://github.com/openshift/monitor-project-lifecycle/issues/24

File: cmd/monitor/monitor.go (hunk 3 above)
Comment (ironcladlou) [reply cmd/monitor/monitor.go:246]: https://github.com/openshift/monitor-project-lifecycle/issues/25

**Pull request merged**